services/src/main/scala/cromwell/pipeline/service/ProjectService.scala

```diff
@@ -34,11 +34,30 @@ class ProjectService(projectRepository: ProjectRepository)(implicit executionCon
     projectRepository.addProject(project)
   }
 
-  def deactivateProjectById(projectId: ProjectId): Future[Option[Project]] =
-    for {
-      _ <- projectRepository.deactivateProjectById(projectId)
-      getProject <- projectRepository.getProjectById(projectId)
-    } yield getProject
+  def deactivateProjectById(projectId: ProjectId, userId: UserId): Future[Option[Project]] = {
+    val result = projectRepository.getProjectById(projectId)
+    result.flatMap {
+      case Some(project) if project.ownerId != userId => Future.failed(new ProjectAccessDeniedException)
+      case Some(_) =>
+        for {
+          _ <- projectRepository.deactivateProjectById(projectId)
+          getProject <- projectRepository.getProjectById(projectId)
+        } yield getProject
+      case None => Future.failed(new ProjectNotFoundException)
+    }
+  }
+
+  def updateProject(request: ProjectUpdateRequest, userId: UserId): Future[Option[Project]] =
+    projectRepository
+      .getProjectById(request.projectId)
+      .flatMap {
+        case Some(project) => if (project.ownerId == userId) {
+          projectRepository.updateProject(project.copy(name = request.name, repository = request.repository))
+          projectRepository.getProjectById(project.projectId)
```

Comment:
It is not guaranteed that you will get a project after update since `projectRepository.updateProject` returns a `Future`.  
It is even possible that `projectRepository.updateProject` will fail, but won't notice it. 
@ilyaonishenko you think that it is "good enough" or just didn't notice it? :)

Reply:
I didn't notice that. 
You're right, this needs to be fixed

Reply:
Discussed today on daily.